thriftbp/circuit_breaker_test.go

```diff
@@ -0,0 +1,106 @@
+package thriftbp_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/reddit/baseplate.go/thriftbp"
+	"github.com/reddit/baseplate.go/thriftbp/thrifttest"
+	"github.com/sony/gobreaker"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	testMinRequests      = 3
+	testFailureThreshold = .5
+	testMethod           = "method"
+)
+
+func TestShouldTrip(t *testing.T) {
+	cb := newTestCircuitBreaker()
+	counts := gobreaker.Counts{
+		Requests:      uint32(testMinRequests + 1),
+		TotalFailures: 0,
+	}
+	tripped := cb.ShouldTripCircuitBreaker(counts)
+	assert.False(t, tripped)
+}
+
+func TestShouldTrip_zeroRequests(t *testing.T) {
```

Comment:
:no_bell: I’m not sure if anyone cares too much but we usually do table driven tests in baseplate.go rather than multiple discrete test functions.

Reply:
Table test makes perfect sense here for combining `TestShouldTrip_zeroRequests`, `TestShouldTrip_failures`, `TestShouldTrip_tooFewRequests`, & `TestShouldTrip_lowFaiureRate`. I'd prefer it, even if just for having more consistent coding style within this repo.

But table tests are also good and you should embrace it :)

Reply:
Changed to table. I think what I dislike about it is how cluttered they usually look -- especially the deep nesting with embedding the t.run() portion. But I guess that is fixable, I like how the code I wrote cleaned up.